packages/@expo/cli/src/start/server/metro/inspector-proxy/handlers/VscodeCompat.ts

```diff
@@ -0,0 +1,121 @@
+    if (message.method === 'Debugger.getPossibleBreakpoints') {
+      const response: DeviceResponse<DebuggerGetPossibleBreakpoints> = {
+        id: message.id,
+        result: { locations: [] },
+      };
+      socket.send(JSON.stringify(response));
+      return true;
+    }
+
+    // Vscode doesn't seem to work nicely with missing `description` fields on `RemoteObject` instances.
+    // See: https://github.com/microsoft/vscode-js-debug/issues/1583
+    if (message.method === 'Runtime.getProperties') {
+      this.interceptGetProperties.add(message.id);
+    }
+
+    // Hermes and vscode have trouble setting breakpoints by `urlRegex` through `Debugger.setBreakpointByUrl`.
+    // Vscode adds `file://` to a URL containing `http://`, which confuses Hermes and sets it to the wrong location.
+    // Hermes needs to create the breakpoint to get the proper ID, but it must be unbounded.
+    // Once the sourcemap is loaded, vscode will rebind the unbounded breakpoint to the correct location (using `Debugger.setBreakpoint`).
+    if (
+      message.method === 'Debugger.setBreakpointByUrl' &&
+      message.params.urlRegex?.startsWith('file:\\/\\/http')
+    ) {
+      // Avoid `urlRegex` to accidentally bind to the wrong location, even when the `url` is invalid
+      message.params.url = message.params.urlRegex;
+      delete message.params.urlRegex;
+    }
+
+    return false;
+  }
+
+  onDeviceMessage(message: DeviceResponse<RuntimeGetProperties> | DeviceRequest<DebuggerPaused>) {
+    // Vscode doesn't seem to work nicely with missing `description` fields on `RemoteObject` instances.
+    // See: https://github.com/microsoft/vscode-js-debug/issues/1583
+    if ('id' in message && this.interceptGetProperties.has(message.id)) {
+      this.interceptGetProperties.delete(message.id);
+
+      for (const item of message.result.result ?? []) {
+        // Force-fully format the properties description to be an empty string
+        // See: https://github.com/facebook/hermes/issues/114
+        if (item.value) {
+          item.value.description = item.value.description ?? '';
+        }
+
+        // Avoid passing the `objectId` for symbol types.
+        // When collapsing in vscode, it will fetch information about the symbol using the `objectId`.
+        // The `Runtime.getProperties` request of the symbol hard-crashes Hermes.
+        if (item.value?.type === 'symbol' && item.value.objectId) {
+          delete item.value.objectId;
+        }
+      }
+    }
+
+    // Hermes adds traces of JSI to the callFrames, which are refering to native code.
+    // It doesn't make sense to show these to the user, so we filter them out.
+    if ('method' in message && message.method === 'Debugger.paused') {
+      message.params.callFrames = message.params.callFrames.filter(
+        (frame) =>
+          frame.location.scriptId !== HERMES_INVALID_SCRIPT_ID &&
+          frame.functionName !== HERMES_NATIVE_FUNCTION_NAME
+      );
```

Comment:
would be good if we could keep at least one frame here for developers to know the crashes is somehow from native side.

Reply:
Do you have an example or some code snippet that triggers the crash? If so, I can limit this filtering to breakpoint `callFrames` only, leaving the crash `callFrames` as is. Does that solve the concern?

Reply:
It's not a required functionality to make it work, so for now I'll just disable this code to not-block the exception stacks.